scripts/check-linters

```diff
@@ -20,7 +20,7 @@ else
 fi
 
 for i in $filelist; do
-    if [[ $i =~ ^3rdparty/ ]]; then
+    if [[ $i =~ ^3rdparty/ || $i =~ ^samples/ ]]; then
```

Comment:
Is there a reason for adding samples here?

Reply:
Yes, the samples/run file has a shellcheck warning it simply because the environment file it sources does not exist (so shellcheck can't check it, and complains). I didn't think we should put `#shellcheck disable` into a sample script, and so excluded them. That said, that means our scripts in samples won't be linted... I guess it'd be fine just to ignore the warning in the script?

Reply:
(My reasoning was that it was exposed to the users, so they shouldn't be worrying about shellcheck.)